src/ui/zcl_abapgit_gui_router.clas.abap

```diff
@@ -713,7 +713,10 @@ CLASS ZCL_ABAPGIT_GUI_ROUTER IMPLEMENTATION.
         lv_xstr = zcl_abapgit_ui_factory=>get_frontend_services( )->file_upload( lv_path ).
         lo_repo->set_files_remote( zcl_abapgit_zip=>load( lv_xstr ) ).
         zcl_abapgit_services_repo=>refresh( lv_key ).
-        ev_state = zcl_abapgit_gui=>c_event_state-re_render.
+        CREATE OBJECT ei_page TYPE zcl_abapgit_gui_page_view_repo
+          EXPORTING
+            iv_key = lo_repo->get_key( ).
+        ev_state = zcl_abapgit_gui=>c_event_state-new_page_replacing.
```

Comment:
I'm a bit doubtful about this part. If you "import" from repo overview that makes sense to switch to specific repo page. But `replacing` means that F3 would exit AG. Apart from expected "back to overview". While if initiated in repo page it should just re-render.
I'd propose to analyse current page to decide on next action. Gui has a feature for this - `ZIF_ABAPGIT_GUI_SERVICES~GET_CURRENT_PAGE_NAME`. So something like `zcl_abapgit_ui_factory=>get_gui_services( )->GET_CURRENT_PAGE_NAME`

P.S. maybe it makes sense to refactor router so that it already have the reference to the GUI instance, cause using factory is a bit fishy, may theoretically lead to infinite loop in some edge cases (not in this one though).

Reply:
Thanks for the feedback, not quite familiar with all the GUI stuff yet 🙂